src/mudlet.cpp

```diff
@@ -4306,3 +4559,501 @@ QPair<bool, QStringList> mudlet::getLines(Host* pHost, const QString& windowName
+    // Ensure the number is at least 1:
+    *ds << qMax(1, wordCount);
+    *ds << QChar(QChar::LineFeed);
+    *ds << wordList.join(QChar::LineFeed).toUtf8();
+    *ds << QChar(QChar::LineFeed);
+    ds->flush();
+    delete ds;
+    if (dictionary.error() != QFile::NoError) {
+        qWarning().nospace().noquote() << "mudlet::prepareDictionary(...) ERROR - failed to completely write dictionary file: \"" << dictionaryPathFileName << "\" status: " << dictionary.errorString();
+        return nullptr;
+    }
+
+    QMultiMap<unsigned int, QString> sortedGraphemeCounts;
+    // Sort the graphemes into a descending order list:
+    if (graphemeCounts.size()) {
+        QHashIterator<QString, unsigned int> itGraphemeCount(graphemeCounts);
+        while (itGraphemeCount.hasNext()) {
+            itGraphemeCount.next();
+            sortedGraphemeCounts.insert(itGraphemeCount.value(), itGraphemeCount.key());
+        }
+    }
+
+    // Generate TRY line:
+    QString tryLine = QStringLiteral("TRY ");
+    QMapIterator<unsigned int, QString> itGrapheme(sortedGraphemeCounts);
+    itGrapheme.toBack();
+    while (itGrapheme.hasPrevious()) {
+        itGrapheme.previous();
+        tryLine.append(itGrapheme.value());
+    }
+
+    affixLines << QStringLiteral("SET UTF-8");
+    affixLines << tryLine;
+
+    // Finally, having got the needed content, write it out:
+    if (!affix.open(QFile::WriteOnly|QFile::Text)) {
+        qWarning().nospace().noquote() << "mudlet::prepareDictionary(...) ERROR - failed to open affix file (for writing): \"" << affixPathFileName << "\" reason: " << affix.errorString();
+        return nullptr;
+    }
+
+    as = new QTextStream(&affix);
+    *as << affixLines.join(QChar::LineFeed).toUtf8();
+    *as << QChar(QChar::LineFeed);
+    as->flush();
+    delete as;
+    if (affix.error() != QFile::NoError) {
+        qWarning().nospace().noquote() << "mudlet::prepareDictionary(...) ERROR - failed to completely write affix file: \"" << affixPathFileName << "\" status: " << affix.errorString();
+        return nullptr;
+    }
+    affix.close();
+
+    // The pair of files are now usable by hunspell library and being use to make
+    // suggestions - they are also capable of being munched - but since we are
+    // using this on our own profiles' dictionaries we will not know the
+    // language that the Mud uses and thus which locale's affixes are suitable.
+
+    // Also, given how we are using the dictionary, any affix rules are going
+    // to confuse our add/remove code.  We just need the SET line to force the
+    // Hunspell API to be UTF-8 and the TRY line to allow for searching for
+    // completions. Anyhow we now need to keep the copy of the word list ourself
+    // to allow for persistant editing of it:
+
+    wordSet = wordList.toSet();
+    return Hunspell_create(affixPathFileName.toUtf8().constData(), dictionaryPathFileName.toUtf8().constData());
+}
+
+Hunhandle* mudlet::prepareSharedDictionary()
```

Comment:
I don't understand what this and the previous function are trying to do. It seems they're trying to do too much at once. Could you split them up into more logical pieces?

Reply:
Does the previous comment explain at all what is going on? If not I can explain in more detail.

Reply:
Ah, given that there are common elements in all of:
* `mudlet::prepareProfileDictionary(...)`
* `mudlet::prepareSharedDictionary()`
* `mudlet::saveDictionary(...)`

I *have* been able to refactor them to call some common smaller methods, in the next commit.